### PR TITLE
Adding dev containers for vscode

### PR DESCRIPTION
**Pull request by eladiw**
## Description
This PR adds a dev containers option for vscode.
DevContainers use containers to create a dev environment which is independent of the user's env and contains all the needed software (dotnet sdk in this case).

This PR is based on docker-compose, which starts one container for the dev and another one for cosmosdb emulator. so once remote containers are started the fhir server is already configured to work with the emulator.

## Related issues
Addresses [issue #].

## Testing
Describe how this change was tested.

## FHIR Team Checklist
- [ ] **Update the title** of the PR to be succinct and less than 50 characters
- [ ] **Add a milestone** to the PR for the sprint that it is merged (i.e. add S47)
- [ ] Tag the PR with the type of update: **Bug**, **Dependencies**, **Enhancement**, or **New-Feature**
- [ ] Tag the PR with **Azure API for FHIR** if this will release to the managed service
- Review [squash-merge requirements](https://github.com/microsoft/fhir-server/blob/master/SquashMergeRequirements.md)

### Semver Change ([docs](https://github.com/microsoft/fhir-server/blob/master/docs/Versioning.md))
Patch|Skip|Feature|Breaking (reason)

### CODE REVIEW

**Review by Ivanidzo4ka — APPROVED** [review]
:shipit:

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	//      1. Start the container
+	//      2. Drag ~/.aspnet/https/aspnetapp.pfx into the root of the file explorer
+	//      3. Open a terminal in VS Code and run "mkdir -p /home/vscode/.aspnet/https && mv aspnetapp.pfx /home/vscode/.aspnet/https"
+	//
+	//    * If only using Remote - Containers with a local container, uncomment this line instead:
+	//      "mounts": [ "source=${env:HOME}${env:USERPROFILE}/.aspnet/https,target=/home/vscode/.aspnet/https,type=bind" ],
+
+	// Use 'postCreateCommand' to run commands after the container is created.
+	// "postCreateCommand": "dotnet restore",
+
```
Comment (Ivanidzo4ka) [inline .devcontainer/devcontainer.json:51]: Is any of these comments necessary?

File: .vscode/launch.json [hunk 2]
```diff
@@ -57,7 +63,8 @@
             "stopAtEntry": false,
             "env": {
                 "ASPNETCORE_ENVIRONMENT": "Development",
-                "DataStore": "CosmosDb"
+                "DataStore": "CosmosDb",
```
Comment (Ivanidzo4ka) [inline .vscode/launch.json:66]: >CosmosDb [](http://example.com/codeflow?start=30&length=8)

should it have cosmosDb host and key as well?

File: .devcontainer/readme.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+The two containers are wired so after switching to the remote container, just hit 'debug/run' (cosmos version) and start debugging.
+
+### Prerequisites
+
+Visual Studio Code
+
+### Usage
+
+1. Install the 'Visual Studio Code Remote - Containers' extension.
+2. vs code will ask to open the folder inside a container. allow it.
```
Comment (Ivanidzo4ka) [inline .devcontainer/readme.md:18]: >vs code will ask to open the folder inside a container. allow it. [](http://example.com/codeflow?start=3&length=65)

super nit:
Capitalization is missing :)

File: .devcontainer/readme.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+3. Your dev env is ready for use.
+
+### Running / Debugging FHIR
+
+1. Run the selected profile
```
Comment (Ivanidzo4ka) [inline .devcontainer/readme.md:23]: >Run the selected profile [](http://example.com/codeflow?start=3&length=24)

Is it worth to add sample how to run specific profile (R4 or STU3)

File: .devcontainer/devcontainer.json (hunk 1 above)
Comment (eladiw) [reply .devcontainer/devcontainer.json:26]: is the comment still relevant after my changes?

File: .vscode/launch.json (hunk 2 above)
Comment (eladiw) [reply .vscode/launch.json:66]: updated

File: .devcontainer/readme.md (hunk 4 above)
Comment (eladiw) [reply .devcontainer/readme.md:31]: added

**Pull request merged**